### ISSUE TITLE
2020-09-02 12:07:23.605391: E tensorflow/compiler/xla/xla_client/tf_logging.cc:11] Check failed: min_shape_dim <= dim && dim<= max_shape_dim

### ISSUE THREAD

**Issue by curehabit** [PyTorchLightning/pytorch-lightning#3318]
## 🐛 Bug

When I try to run mnist tutorial (https://pytorch-lightning.readthedocs.io/en/stable/new-project.html), I got an error:

Value out of range (expected to be in range of [-1, 0], but got 1)
2020-09-02 12:07:23.605391: E tensorflow/compiler/xla/xla_client/tf_logging.cc:11] Check failed: min_shape_dim <= dim && dim<= max_shape_dim

My device is torch_xla, and I set Trainer( tpu_cores=1).

### To Reproduce

Just run as the tutorial.

```
pytorch_lightning/core/step_result.py(844)weighted_mean()
-> weights = weights.to(result.device)
(Pdb) l
839             for k, v in predictions_dict.items():
840                 self.write(k, v, filename)
841
842
843     def weighted_mean(result, weights):
844 B->     weights = weights.to(result.device)
845         numerator = torch.dot(result.float(), weights.t().float()) ### Broken there
846         result = numerator / weights.sum().float()
847         return result

Value out of range (expected to be in range of [-1, 0], but got 1)
2020-09-02 12:07:23.605391: E tensorflow/compiler/xla/xla_client/tf_logging.cc:11] Check failed: min_shape_dim <= dim && dim<= max_shape_dim
*** Begin stack trace ***
        tensorflow::CurrentStackTrace()
        torch_xla::XlaHelpers::GetCanonicalDimensionIndex(long long, long long)
        torch_xla::XlaHelpers::MakeTransposePermutation(long long, long long, long long)
        torch_xla::XLATensor::transpose(torch_xla::XLATensor const&, long long, long long)
        torch_xla::AtenXlaType::t(at::Tensor const&)
        c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoRuntimeFunctor_<at::Tensor (*)(at::Tensorconst&), at::Tensor, c10::guts::typelist::typelist<at::Tensor const&> >, at::Tensor (at::Tensor const&)>::call(c10::OperatorKernel*, at::Tensor const&)
        at::Tensor c10::Dispatcher::call<at::Tensor, at::Tensor const&>(c10::TypedOperatorHandle<at::Tensor (at::Tensor const&)> const&, at::Tensor const&) const
        at::t(at::Tensor const&)
```

### Environment

* CUDA:
        - GPU:
        - available:         False
        - version:           None
* Packages:
        - numpy:             1.19.0
        - pyTorch_debug:     False
        - pyTorch_version:   1.6.0.dev20200622
        - pytorch-lightning: 0.9.0
        - tensorboard:       2.2.0
        - tqdm:              4.48.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                -
        - processor:
        - python:            3.7.7
        - version:           #1 SMP Debian 4.14.81.bm.15 Sun Sep 8 05:02:31 UTC 2019

 - PyTorch Version (e.g., 1.0): 1.6
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7.8
 - CUDA/cuDNN version: None
 - GPU models and configuration: None
 - Any other relevant information: torch_xla:1.6.0

### Additional context

I try to use pdb to debug it, and I found weights.t() caused this error. 
```
def weighted_mean(result, weights):
844 B->     weights = weights.to(result.device)
845         numerator = torch.dot(result.float(), weights.t().float()) ### Broken there
846         result = numerator / weights.sum().float()
847         return result
(Pdb) weights.shape
torch.Size([1])
(Pdb) result.shape
torch.Size([1])
(Pdb) weights.t
<built-in method t of Tensor object at 0x7fd0142f6be0>
(Pdb) weights.t()
2020-09-02 12:07:23.605391: E tensorflow/compiler/xla/xla_client/tf_logging.cc:11] Check failed: min_shape_dim <= dim && dim<= max_shape_dim
```

if I set weights = weights.t() before weights.to(result.device), it can run normally.
```
def weighted_mean(result, weights):
    weights = weights.t().to(result.device) # weights = weights.to(result.device)
    numerator = torch.dot(result.float(), weights.float()) 
    result = numerator / weights.sum().float()
    return result
```

**Comment by tfrizza** [comment]
Also had this issue when using TPU +1

**Comment by Borda** [comment]
@lezwon was working on this one, it shall be fixed #3252, mind trying master
